main.py

```diff
@@ -0,0 +1,83 @@
+import os
+from typing import Any, Dict
+
+from dotenv import load_dotenv
+from fastapi import FastAPI, HTTPException
+from openai import OpenAI
+from pydantic import BaseModel
+
+from pseudoanonymize.anonymization import Anonymizer
+from pseudoanonymize.deanonymization import Deanonymizer
+from pseudoanonymize.exceptions import MaxRetriesExceededException
+
+app = FastAPI()
+
+load_dotenv()
+openai_api_key = os.getenv("OPENAI_API_KEY")
+client = OpenAI(api_key=openai_api_key)
+
+
+anonymizer = Anonymizer(client=client)
+deanonymizer = Deanonymizer(client=client)
+
+
+class AnonymizeRequest(BaseModel):
+    text: str
+    retries: int = 5
+
+
+class DeanonymizeRequest(BaseModel):
+    text: str
+    replacement_dict: Dict[Any, str]
+    retries: int = 5
+
+
+def flatten_replacement_dict(replacement_dict):
+    flat_dict = {}
+    for key, value in replacement_dict.items():
+        if isinstance(key, tuple):
+            for sub_key in key:
+                flat_dict[sub_key] = value
+        else:
+            flat_dict[key] = value
+    return flat_dict
+
+
+@app.post("/anonymize")
+async def anonymize(request: AnonymizeRequest):
+    try:
+        prediction = anonymizer.retry_prediction({"text": request.text}, request.retries)
+        flattened_dict = flatten_replacement_dict(prediction["replacement_dict"])
+        return {"anonymized_text": prediction["anonymized_text"], "replacement_dict": flattened_dict}
```

Comment:
Create a Pydantic model called `AnonymizeResponse` to model this output.

Benefits:
- Someone reading the code can clearly see the returned structure.
- Makes response format readable for the autogenerated API docs.
- Less bug prone due to strict type checks

Reply:
Done